pkg/generate/app/cmd/newapp.go

```diff
@@ -334,6 +334,14 @@ func (c *AppConfig) addReferenceBuilderComponents(b *app.ReferenceBuilder) {
 		input.Searcher = searcher
 		return input
 	})
+
+	_, repos, _ := b.Result()
+	for _, repo := range repos {
+		repo.SetContextDir(c.ContextDir)
+		if c.Strategy == "docker" {
```

Comment:
we could reuse the api one, but in new-app code we don't use a constant :-( 
given this could potentially go in, I'd like to keep my changes as low-impact as possible :)